pkgs/development/python-modules/pycategories/default.nix

```diff
@@ -0,0 +1,32 @@
+{ buildPythonPackage
+, pytestcov
+, fetchPypi
+, lib
+, pytest
+, pythonOlder
+, pytestrunner
+}:
+
+buildPythonPackage rec {
+  pname = "pycategories";
+  version = "1.1.0";
+  disabled = pythonOlder "3.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1vkza285jr2qzzq6bnyas93xx6my8rn3smc3vsdwa6kxx660l96w";
+  };
+
+  nativeBuildInputs = [ pytestrunner ];
+  buildInputs = [ (import ./infix.nix { inherit buildPythonPackage lib fetchPypi; }) ];
```

Comment:
I would just add this to  pkgs/top-level/python-packages.nix and treat it like another package. In a separate commit of course :).

Reply:
I think we should be using `callPackage` for this and also, why `buildInputs`? Usually it would be `propagatedBuildInputs`.

Reply:
didn't even notice that. Yes, propagatedBuildInputs would be the more correct choice because in python, you're likely going to need access to those dependencies again.

Reply:
I am now adding `infix` via `callPackage` and added it and `pytestrunner` via `propagatedBuildInputs`.